### DISCUSSION TITLE
What's the Best Practice to manage sets of constants conditionally? (.env mode?)

### DISCUSSION THREAD

**Discussion by shkim** [remix-run/remix#4752] [Q&A] [answered]
It seems that remix.run reads constants from `.env` file only.
I could get values such as API Key, API Target Host URL, Cookie Secret, App Version, etc from .env file.

Now I wonder whether I can conditionally load these values by `mode` or some other parameter.
So I'd like to manage multiple set of .env files, like: `.env.dev`, `.env.beta`, `.env.production`

**Comment by sergiodxa** [comment, marked as answer]
I would recommend you to use `.env` as it was originally intended only for local development.

For production, use what your server provides you to handle env variables and secrets, for example Vercel, Netlify, Cloudflare, etc. all have a place where you can configure your production and staging secrets and env variables.

For CI, as far as I remember all services (GitHub Actions, CircleCI, TravisCI, etc.) have also a place to define those env variables.

But if you still want to use different `.env` files for dev, prod, etc. you can switch from Remix App Server to Express, then you'll have full control of your server and could run `dotenv` package against different `.env` files.
